src/stdlib_sparse_spmv.fypp

```diff
@@ -529,7 +529,7 @@ contains
             integer :: j, k
             do j = 1, n
```

Comment:
would it be suitable to replace the inner loops with an array operations?
```
y(col(:,j)) = y(col(:,j)) + alpha_ * a(:,j) * x(:)
```

Reply:
unfortunatelly no, it can happen that `col(:,j)` will request a jump-back to a "previous" index. When such access is on the lhs of the assigment with max optimization it will happen that the second request to a previous index will not happen and the result will be incorrect.
I had alreay seen this before and tested it again with the unit test, indeed if I try to use array expression instead of a plain do-loop the test fails because of that.